common/src/main/java/com/rtbhouse/grpc/loadbalancer/HealthCheckCapableGrpcServerBase.java

```diff
@@ -0,0 +1,109 @@
+package com.rtbhouse.grpc.loadbalancer;
+
+import io.grpc.BindableService;
+import io.grpc.ManagedChannel;
+import io.grpc.Server;
+import io.grpc.health.v1.HealthCheckRequest;
+import io.grpc.health.v1.HealthCheckResponse;
+import io.grpc.health.v1.HealthCheckResponse.ServingStatus;
+import io.grpc.health.v1.HealthGrpc;
+import io.grpc.health.v1.HealthGrpc.HealthBlockingStub;
+import io.grpc.health.v1.HealthGrpc.HealthImplBase;
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+import java.io.IOException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class HealthCheckCapableGrpcServerBase extends GrpcServerBase {
```

Comment:
Po co jest ta klasa? Czy healthchecka nie da się zrobić jako niezależnego komponentu? Jeśli nie to po co jest osobna klasa dodające healthchecka? Czy nie może to być w GrpcServerBase? I tak jeśli chce się użyć, którąś z gotowych implementacji to one i tak dziedziczą po tej klasie.

Reply:
Ta obsługa healthchecka jest jeszcze ogólnie do przedyskutowania, nie jestem pewien czy dobrze to robimy że teraz użytkownik daje nam coś typu HealthGrpc.HealthImplBase czyli implementację googlowego serwisu HealthProto, a my stawiamy z tym InProcessServer i odpytujemy go. Na razie zostawiłem tak jak jest, zobaczymy co jeszcze inni powiedzą.